azure/services/managedclusters/spec.go

```diff
@@ -414,6 +432,15 @@ func (s *ManagedClusterSpec) Parameters(ctx context.Context, existing interface{
 		}
 	}
 
+	if s.HTTPProxyConfig != nil {
+		managedCluster.HTTPProxyConfig = &containerservice.ManagedClusterHTTPProxyConfig{
```

Comment:
Now that you mention it, I see how it would be awkward to use an empty string as the uninitialized value, since we'd have `HTTPProxy = ptr.To("")` instead of `nil.`

Reply:
yep